From 35f169f75fbe01f2c80bd72ebb231697023919e3 Mon Sep 17 00:00:00 2001
From: Cool Lee <cool_lee@aspeedtech.com>
Date: Thu, 20 Nov 2025 11:53:17 +0800
Subject: [PATCH] dts: ast2700-ibex-reserved-mem: Refine reserved memory

- Introduce `EDAF_128MB` macro to support 128MB or 64MB eDAF DDR mode.
- Adjust reserved memory regions for eDAF, eSPI, PCIe, and SSP based on
  configuration flags (`DUAL_NODE`, `PCIE0_EP`, `PCIE1_EP`).
- Consolidate and reorganize memory addresses to align with updated
  hardware mapping.

Signed-off-by: Cool Lee <cool_lee@aspeedtech.com>
Change-Id: I6fa78384af5dc9012fdc3170f16acf3eb623161c
(cherry picked from commit 350fcabf1c69da2d4e4a776f4dec9169ea834901)
Signed-off-by: Vasanthakumar K <vasanthakumark@ami.com>
---
 arch/riscv/dts/ast2700-ibex-reserved-mem.dtsi | 81 +++++++++++--------
 arch/riscv/dts/ast2700-ibex.dts               |  1 +
 2 files changed, 50 insertions(+), 32 deletions(-)

diff --git a/arch/riscv/dts/ast2700-ibex-reserved-mem.dtsi b/arch/riscv/dts/ast2700-ibex-reserved-mem.dtsi
index cd619881dc5b..2804c3c61895 100644
--- a/arch/riscv/dts/ast2700-ibex-reserved-mem.dtsi
+++ b/arch/riscv/dts/ast2700-ibex-reserved-mem.dtsi
@@ -2,80 +2,97 @@
 
 /* AST2700 reserved memories with no-map property */
 
-#if PCIE0_EP
-bmc_dev0_memory: bmc-dev0-memory@9b800000 {
-	reg = <0x9b800000 0x100000>;
+#if DUAL_NODE
+espi1_mmbi_memory: espi1-mmbi-memory@93800000 {
+	reg = <0x93800000 0x4000000>;
 	no-map;
 	bootph-all;
 };
 
-xdma_memory0: xdma0 {
-	size = <0x0 0x00100000>;
-	compatible = "shared-dma-pool";
+#if EDAF_128MB
+// This area used for eDAF DDR mode even in MIX mode or HW mode(enabled in SPL)
+edaf1: edaf1-memory@97800000 {
+	reg = <0x97800000 0x8000000>;
 	no-map;
+	bootph-all;
 };
-
-pcie0_mmbi0_memory: pcie0-mmbi0-memory@9ba00000 {
-	reg = <0x9ba00000 0x200000>;
+#else
+edaf1: edaf1-memory@9b800000 {
+	reg = <0x9b800000 0x4000000>;
 	no-map;
 	bootph-all;
 };
 #endif
+#endif
 
-#if PCIE1_EP
-bmc_dev1_memory: bmc-dev1-memory@9b900000 {
-	reg = <0x9b900000 0x100000>;
+#if EDAF_128MB
+// This area used for eDAF DDR mode even in MIX mode or HW mode(enabled in SPL)
+edaf0: eadf0-memory@49f800000 {
+	reg = <0x9f800000 0x8000000>;
 	no-map;
 	bootph-all;
 };
-
-xdma_memory1: xdma1 {
-	size = <0x0 0x00100000>;
-	compatible = "shared-dma-pool";
+#else
+edaf0: eadf0-memory@a3800000 {
+	reg = <0xa3800000 0x4000000>;
 	no-map;
+	bootph-all;
 };
+#endif
 
-pcie1_mmbi4_memory: pcie1-mmbi4-memory@9bc00000 {
-	reg = <0x9bc00000 0x200000>;
+espi0_mmbi_memory: espi0-mmbi-memory@a7800000 {
+	reg = <0xa7800000 0x4000000>;
 	no-map;
 	bootph-all;
 };
-#endif
 
-pcie2_mmbi0_memory: pcie2-mmbi0-memory@9be00000 {
-	reg = <0x9be00000 0x200000>;
+#if PCIE0_EP
+bmc_dev0_memory: bmc-dev0-memory@ab800000 {
+	reg = <0xab800000 0x100000>;
 	no-map;
 	bootph-all;
 };
 
-espi0_mmbi_memory: espi0-mmbi-memory@9c000000 {
-	reg = <0x9c000000 0x4000000>;
+xdma_memory0: xdma0 {
+	size = <0x0 0x00100000>;
+	compatible = "shared-dma-pool";
+	no-map;
+};
+
+pcie0_mmbi0_memory: pcie0-mmbi0-memory@aba00000 {
+	reg = <0xaba00000 0x200000>;
 	no-map;
 	bootph-all;
 };
+#endif
 
-// This area used for eDAF DDR mode even in MIX mode or HW mode(enabled in SPL)
-edaf0: eadf0-memory@a0000000 {
-	reg = <0xa0000000 0x4000000>;
+#if PCIE1_EP
+bmc_dev1_memory: bmc-dev1-memory@ab900000 {
+	reg = <0xab900000 0x100000>;
 	no-map;
 	bootph-all;
 };
 
-#if DUAL_NODE
-espi1_mmbi_memory: espi1-mmbi-memory@a4000000 {
-	reg = <0xa4000000 0x4000000>;
+xdma_memory1: xdma1 {
+	size = <0x0 0x00100000>;
+	compatible = "shared-dma-pool";
 	no-map;
 	bootph-all;
 };
 
-// This area used for eDAF DDR mode even in MIX mode or HW mode(enabled in SPL)
-edaf1: edaf1-memory@a8000000 {
-	reg = <0xa8000000 0x4000000>;
+pcie1_mmbi4_memory: pcie1-mmbi4-memory@abc00000 {
+	reg = <0xabc00000 0x200000>;
 	no-map;
 	bootph-all;
 };
 #endif
 
+pcie2_mmbi0_memory: pcie2-mmbi0-memory@abe00000 {
+	reg = <0xabe00000 0x200000>;
+	no-map;
+	bootph-all;
+};
+
 ssp_memory: ssp-memory@ac000000 {
 	reg = <0xac000000 0x2000000>;
 	no-map;
diff --git a/arch/riscv/dts/ast2700-ibex.dts b/arch/riscv/dts/ast2700-ibex.dts
index 7553debbca26..9f0531af4c6f 100644
--- a/arch/riscv/dts/ast2700-ibex.dts
+++ b/arch/riscv/dts/ast2700-ibex.dts
@@ -5,6 +5,7 @@
 #include "ast2700.dtsi"
 
 #define DUAL_NODE 0	// 1: DUAL_NODE, 0: SINGLE_NODE
+#define EDAF_128MB 0	// 1: 128MB, 0: 64MB
 #define PCIE0_EP 1	// 1: EP, 0: RC
 #define PCIE1_EP 1	// 1: EP, 0: RC
 #define PCIE2_RC 1	// 1: RC, 0: SGMII
-- 
2.34.1

